plugin/goto.py

```diff
@@ -40,22 +40,72 @@ def run(self, edit: sublime.Edit, event: 'Optional[dict]' = None) -> None:
                     request, lambda response: self.handle_response(response, pos))
 
     def handle_response(self, response: 'Optional[Any]', position: int) -> None:
+        def process_response_list(responses: list) -> 'List[Tuple[str, str, Tuple[int, int]]]':
+            return [process_response(x) for x in responses]
+
+        def process_response(response: dict) -> 'Tuple[str, str, Tuple[int, int]]':
+            if "targetUri" in response:
+                # TODO: Do something clever with originSelectionRange and targetRange.
+                file_path = uri_to_filename(response["targetUri"])
+                start = Point.from_lsp(response["targetSelectionRange"]["start"])
+            else:
+                file_path = uri_to_filename(response["uri"])
+                start = Point.from_lsp(response["range"]["start"])
+            row = start.row + 1
+            col = start.col + 1
+            file_path_and_row_col = "{}:{}:{}".format(file_path, row, col)
+            return file_path, file_path_and_row_col, (row, col)
+
+        def open_location(window: sublime.Window, location: 'Tuple[str, str, Tuple[int, int]]') -> None:
+            fname, file_path_and_row_col, rowcol = location
+            row, col = rowcol
+            debug("opening location", file_path_and_row_col)
+            window.open_file(
+                file_path_and_row_col,
+                sublime.ENCODED_POSITION | sublime.FORCE_GROUP)
+
+        def select_entry(
+                window: sublime.Window,
+                locations: 'List[Tuple[str, str, Tuple[int, int]]]',
+                idx: int,
+                orig_view: sublime.View) -> None:
+            if idx >= 0:
+                open_location(window, locations[idx])
+            else:
+                if orig_view:
+                    orig_view.sel().clear()
```

Comment:
orig_view.sel().clear() removes all cursors. We've had some comments in the past that this is undesirable. Do we even need to call orig_view.sel().clear()? Perhaps we can just leave it out. Otherwise, we should add back the original selection. The choice is up to you.

Reply:
For simplicity I removed the selection clear, but ideally it would restore the original selection.  The only thing is that mypy falsely complains that sublime.Selection isn't iterable.  (looks like its a known issue in mypy https://github.com/python/mypy/issues/2220)  So I'm not sure if there's a nice way to ignore the error?

Reply:
You may use `# type: ignore` after an expression/statement to make mypy ignore that line.